src/state/internal/createQueryStore.ts

```diff
@@ -0,0 +1,999 @@
+  },
+  customStateCreator: StateCreator<U, [], [['zustand/subscribeWithSelector', never]]>,
+  persistConfig?: RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>
+): QueryStore<TData, TParams, StoreState<TData, TParams> & U>;
+
+export function createQueryStore<
+  TQueryFnData,
+  TParams extends Record<string, unknown> = Record<string, unknown>,
+  U = unknown,
+  TData = TQueryFnData,
+>(
+  config: QueryStoreConfig<TQueryFnData, TParams, TData, StoreState<TData, TParams> & PrivateStoreState & U> & {
+    params?: { [K in keyof TParams]: ParamResolvable<TParams[K], TParams, StoreState<TData, TParams> & PrivateStoreState & U, TData> };
+  },
+  persistConfig?: RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>
+): QueryStore<TData, TParams, StoreState<TData, TParams> & U>;
+
+/**
+ * Creates a query-enabled Rainbow store with data fetching capabilities.
+ * @template TQueryFnData - The raw data type returned by the fetcher
+ * @template TParams - Parameters passed to the fetcher function
+ * @template U - User-defined custom store state
+ * @template TData - The transformed data type, if applicable (defaults to `TQueryFnData`)
+ */
+export function createQueryStore<
+  TQueryFnData,
+  TParams extends Record<string, unknown> = Record<string, unknown>,
+  U = unknown,
+  TData = TQueryFnData,
+>(
+  config: QueryStoreConfig<TQueryFnData, TParams, TData, StoreState<TData, TParams> & PrivateStoreState & U> & {
+    params?: { [K in keyof TParams]: ParamResolvable<TParams[K], TParams, StoreState<TData, TParams> & PrivateStoreState & U, TData> };
+  },
+  arg1?:
+    | StateCreator<U, [], [['zustand/subscribeWithSelector', never]]>
+    | RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>,
+  arg2?: RainbowPersistConfig<StoreState<TData, TParams> & PrivateStoreState & U>
+): QueryStore<TData, TParams, StoreState<TData, TParams> & U> {
+  type S = StoreState<TData, TParams> & PrivateStoreState & U;
+
+  /* If arg1 is a function, it's the customStateCreator; otherwise, it's the persistConfig. */
+  const customStateCreator = typeof arg1 === 'function' ? arg1 : () => ({}) as U;
+  const persistConfig = typeof arg1 === 'object' ? arg1 : arg2;
+
+  const {
+    abortInterruptedFetches = true,
+    fetcher,
+    onFetched,
+    transform,
+    cacheTime = time.days(7),
+    debugMode = false,
+    disableAutoRefetching = false,
+    disableCache = false,
+    enabled = true,
+    keepPreviousData = false,
+    maxRetries = 3,
+    onError,
+    params,
+    retryDelay = time.seconds(5),
+    setData,
+    staleTime = time.minutes(2),
+    suppressStaleTimeWarning = false,
+  } = config;
+
+  if (IS_DEV && !suppressStaleTimeWarning && staleTime < MIN_STALE_TIME) {
+    console.warn(
+      `[createQueryStore${persistConfig?.storageKey ? `: ${persistConfig.storageKey}` : ''}] ❌ Stale times under ${
+        MIN_STALE_TIME / 1000
+      } seconds are not recommended.`
+    );
+  }
```

Comment:
I wonder if we should just override the value if it's beneath this value. I don't know if you see a reason for having it exist, but seems to me that it should be disallowed.

Reply:
You might want to create a query that fetches freshest data with `staleTime = 0`, but that doesn't need to be refetched on an interval by setting `disableAutoRefetching`